### PR TITLE
Allow new charts to generate sums

### PR DESCRIPTION
**Pull request by aiyengar2**
Related Issue: https://github.com/rancher/rancher/issues/29173

Dependent on merge:
- [ ] Regenerates charts that merged with bug in generator (https://github.com/rancher/charts/pull/702#pullrequestreview-496872722)

### CODE REVIEW

**Comment by aiyengar2** [comment]
We'll need to port these changes over to `partner-charts` and `rke-charts` as well

File: scripts/produce-sha256 [hunk 1]
```diff
@@ -16,14 +16,16 @@ for f in packages/*; do
...
     cp -R sha256sum-new sha256sum
     for f in packages/*; do
         echo $(basename -- ${f})
     done
 fi
 
-for f in sha256sum/*; do 
+# For all the new sha256sums that were generated, compare them and see if the existing assets need to be rebuilt
+for f in sha256sum-new/*; do
     if ! cmp -s ./sha256sum-new/$(basename -- ${f})/$(basename -- ${f}).sum ${f}/$(basename -- ${f}).sum; then
```
Comment (cbron) [inline scripts/produce-sha256:29]: I think you have to swap this too, you are comparing same file here

Comment (aiyengar2) [reply scripts/produce-sha256:29]: good catch!

**Review by StrongMonkey — APPROVED** [review]
Whoops, good catch!

**Comment by aiyengar2** [comment]
@StrongMonkey do you want to test this out locally before we merge it? Unfortunately I can't run the script since `sha256sum` isn't available on Macs.

**Comment by StrongMonkey** [comment]
@aiyengar2 Looks good with your changes!

```
➜  charts git:(fix-ci) ./scripts/produce-sha256
backup-restore-operator
fleet
fleet-agent
fleet-crd
rancher-cis-benchmark
rancher-gatekeeper
rancher-istio
rancher-kiali-server
rancher-logging
rancher-monitoring
rancher-operator
rancher-operator-crd
rancher-pushprox
rancher-webhook
rio
➜  charts git:(fix-ci) ✗ mkdir -p packages/testing           
➜  charts git:(fix-ci) ✗ echo "hello" > packages/testing/file
➜  charts git:(fix-ci) ✗ ./scripts/produce-sha256            
testing
➜  charts git:(fix-ci) ✗ 
```

**Review by cbron — APPROVED** [review]

**Pull request merged**